commit 555d3d5d2be13675490a80df0d7961551822ef1f
Author: Vlad Yasevich <vladislav.yasevich@hp.com>
Date:   Thu Nov 29 08:56:16 2007 -0500

    SCTP: Fix chunk acceptance when no authenticated chunks were listed.
    
    In the case where no autheticated chunks were specified, we were still
    trying to verify that a given chunk needs authentication and doing so
    incorrectly.  Add a check for parameter length to make sure we don't
    try to use an empty auth_chunks parameter to verify against.
    
    Signed-off-by: Vlad Yasevich <vladislav.yasevich@hp.com>

diff --git a/net/sctp/auth.c b/net/sctp/auth.c
index 6d5fa6bb371b..6d89e35307aa 100644
--- a/net/sctp/auth.c
+++ b/net/sctp/auth.c
@@ -631,7 +631,7 @@ static int __sctp_auth_cid(sctp_cid_t chunk, struct sctp_chunks_param *param)
 	int found = 0;
 	int i;
 
-	if (!param)
+	if (!param || param->param_hdr.length == 0)
 		return 0;
 
 	len = ntohs(param->param_hdr.length) - sizeof(sctp_paramhdr_t);